k9mail/src/main/java/com/fsck/k9/activity/MessageReference.java

```diff
@@ -1,94 +1,63 @@
 package com.fsck.k9.activity;
 
-import android.content.Context;
-import android.os.Parcel;
-import android.os.Parcelable;
-import android.util.Log;
-
-import com.fsck.k9.Account;
-import com.fsck.k9.K9;
-import com.fsck.k9.Preferences;
+
+import java.util.StringTokenizer;
+
+import android.support.annotation.Nullable;
+
 import com.fsck.k9.mail.Flag;
-import com.fsck.k9.mail.MessagingException;
-import com.fsck.k9.mailstore.LocalFolder;
-import com.fsck.k9.mailstore.LocalMessage;
 import com.fsck.k9.mail.filter.Base64;
 
-import java.util.StringTokenizer;
+import static com.fsck.k9.helper.Preconditions.checkNotNull;
+
+
+public class MessageReference {
+    private static final char IDENTITY_VERSION_1 = '!';
```

Comment:
Well this is a hideous way of versioning (I know you didn't introduce this code so I'm not asking for it to be fixed).